gc/base/MemoryPool.hpp

```diff
@@ -346,6 +346,49 @@ class MM_MemoryPool : public MM_BaseVirtual
 
 	MMINLINE virtual uintptr_t getDarkMatterSamples() { return _darkMatterSamples; }
 
+	MMINLINE virtual uintptr_t getFreeMemoryAndDarkMatterBytes() {
+		return getActualFreeMemorySize() + getDarkMatterBytes();
+	}
+
+	/**
+	 * Update memory pool statistical data
+	 *
+	 * @param freeBytes free bytes added
+	 * @param freeEntryCount free memory elements added
+	 * @param largestFreeEntry largest free memory element size
+	 */
+	MMINLINE void updateMemoryPoolStatistics(MM_EnvironmentBase *env, uintptr_t freeBytes, uintptr_t freeEntryCount, uintptr_t largestFreeEntry)
+	{
+		setFreeMemorySize(freeBytes);
+		setFreeEntryCount(freeEntryCount);
+		setLargestFreeEntry(largestFreeEntry);
+	}
+
+	virtual void recalculateMemoryPoolStatistics(MM_EnvironmentBase* env)
+	{
+		Assert_MM_unreachable();
+	}
+	virtual bool recycleHeapChunk(void* chunkBase, void* chunkTop)
+	{
+		Assert_MM_unreachable();
+		return false;
+	}
+	virtual void fillWithHoles(void *addrBase, void *addrTop)
+	{
+		Assert_MM_unreachable();
+	}
+	MMINLINE virtual uintptr_t getScannableBytes()
+	{
+		Assert_MM_unreachable();
+		return 0;
+	}
+	MMINLINE virtual uintptr_t getNonScannableBytes()
+	{
+		Assert_MM_unreachable();
+		return 0;
+	}
```

Comment:
Everything about scannable bytes (all APIs and fields) could be here in base class, and no need for methods to be virtual.

I'd move also increment methods as well.

Reply:
incrementScannableBytes() also have different implementation in MM_MemoryPoolBumpPointer.

Reply:
hm, do we call incrementScannableBytes anymore (we should)? how come you did not introduce one is Base or AOL?

Reply:
I have moved incrementScannableBytes(), getScannableBytes(), getNonScannableBytes() and related fields to Base as you suggested, we call it from MM_SweepPoolManagerAddressOrderedListBase and in MM_SweepPoolManagerVLHGC(old openj9).

Reply:
oops, forgot removing it from AOLBase

Reply:
ok, looks good now. 
please test that it compiles/runs ok as a stand alone change too.

Reply:
ok, I will launch two personal builds(included one for stand alone change only).